File: ports/csharp/src/core/thread/BcsMutex.cs
using System.Threading;

namespace BCS.Core.Thread {
    // BcsMutex maps to C# Monitor/lock semantics
    public class BcsMutex {
        private readonly object _lockObj = new object();

        public void Lock() {
            Monitor.Enter(_lockObj);
        }

        public void Unlock() {
            Monitor.Exit(_lockObj);
        }
    }
}

File: ports/go/src/core/thread/bcsmutex.go
package thread

import "sync"

// BcsMutex maps exactly to Go's sync.Mutex
type BcsMutex struct {
	mu sync.Mutex
}

func (m *BcsMutex) Lock() {
	m.mu.Lock()
}

func (m *BcsMutex) Unlock() {
	m.mu.Unlock()
}
